### PR TITLE
react-relay - improve props and relay variables typing

### PR DESCRIPTION
**Pull request by npirotte**
- Adds generic type for RelayProps to type variables.
- Adds RelayProps interface that can be used to fully abstract relay prop injection.

```ts
interface IProps extends Relay.RelayProps<ComponentRelayVariables> {
  // Everything but no relay.
}
```

### CODE REVIEW

File: types/react-relay/classic.d.ts [hunk 1]
```diff
@@ -209,13 +209,17 @@ export type OnReadyStateChange = (
     }
 ) => void;
 
-export interface RelayProp {
+export interface RelayProp<V> {
```
Comment (alloy) [inline types/react-relay/classic.d.ts]: Can you make this default to `any` so it stays backwards compatible?

Comment (npirotte) [reply types/react-relay/classic.d.ts]: Indeed, good idea.

Comment (npirotte) [reply types/react-relay/classic.d.ts]: That done.

**Comment by alloy** [comment]
Thanks!

**Pull request merged**